### PR TITLE
[MGDOBR-377] Run integration-tests in parallel

### PR DESCRIPTION
**Pull request by radtriste**
[MGDOBR-377](https://issues.redhat.com/browse/MGDOBR-377) 

Please make sure that your PR meets the following requirements:

- [ ] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config)
- [ ] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-1] - $clear_explanation_of_what_you_did"
- [ ] All new functionality is tested
- [ ] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

<details>
<summary>
How to trigger pipelines and use the bots:
</summary>

* <b>Run the end to end pipeline</b>  
  Annotate the pull request with the label: `safe to test`. If you want to run the pipeline again, remove and add it again.

* <b>Rebase the pull request</b>  
  Comment with: `/rebase`.

</details>

### CODE REVIEW

File: integration-tests/pom.xml [hunk 1]
```diff
@@ -13,6 +13,7 @@
   <artifactId>integration-tests</artifactId>
   <version>0.0.1-SNAPSHOT</version>
   <name>Event Bridge :: Integration Tests</name>
+
```
Comment (jiripetrlik) [inline integration-tests/pom.xml:16]: Please remove.

File: .github/workflows/Manager-CI.yaml [hunk 2]
```diff
@@ -5,6 +5,7 @@ on:
     branches:    
       - 'main'
     paths:
+      - '.github/workflows/Manager-CI.yaml'
```
Comment (jiripetrlik) [inline .github/workflows/Manager-CI.yaml:8]: Can you explain why we need this path?

File: integration-tests/pom.xml (hunk 1 above)
Comment (radtriste) [reply integration-tests/pom.xml:16]: This was to check that the parallel testing does work on `Manager - CI / Build all images and run E2E tests (pull_request_target)`. Else the check does not trigger. Gonna remove that.

File: .github/workflows/Manager-CI.yaml (hunk 2 above)
Comment (radtriste) [reply .github/workflows/Manager-CI.yaml:8]: Manager-CI check should be launched also when `.github/workflows/Manager-CI.yaml'` is modified.

**Comment by radtriste** [comment]
> @radtriste Looks like the tests in PR checks are not executed in parallel. Maybe we can specify fixed parallelism with parallelism lets say 3?
> 
> https://github.com/cucumber/cucumber-jvm/tree/main/junit-platform-engine

Trying https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/399/commits/00442fab9df2f63fc15bd4997d39492dcfc7c963, let's see ...

File: shard-operator-integration-tests/pom.xml [hunk 3]
```diff
@@ -16,6 +16,7 @@
 
   <properties>
     <parallel>false</parallel>
+    <parallel.concurrency.number>3</parallel.concurrency.number>
```
Comment (jomarko) [inline shard-operator-integration-tests/pom.xml:19]: We should do the same change also in `integration-tests/pom.xml`?

**Review by jomarko — COMMENTED** [review]
One question

**Comment by sutaakar** [comment]
@radtriste Looks better now, just try to add also `cucumber.junit-platform.naming-strategy=long` to be able to see feature name for every step in log.

**Review by jiripetrlik — APPROVED** [review]
@radtriste 
Thank you for your reply to my questions. LGTM

**Comment by radtriste** [comment]
waiting for https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/430

**Review by jomarko — APPROVED** [review]

**Comment by radtriste** [comment]
@sutaakar should we merge that one ? Are we stable enough for parallel testing ? :D

**Comment by sutaakar** [comment]
@radtriste I still see some issues in periodic tests - would like to analyze/address that first:
https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/actions/workflows/dev-cluster-check.yml

**Comment by radtriste** [comment]
ok. will put on hold. let me know. or merge when you want as I will be on PTO ;)

**Review by sutaakar — APPROVED** [review]

**Pull request merged**